Network.hs

```diff
@@ -292,6 +293,7 @@ accept sock@(MkSocket _ AF_INET6 _ _ _) = do
 # if !defined(mingw32_HOST_OS)
                  SockAddrUnix      a   -> return a
 # endif
+                 _ -> error "accept: socket address not supported on this platform."
```

Comment:
I'm dissatisfied with these catch alls, but it is an improvement over pattern match failures. CPP makes it quite ugly to be total here. I'd rather be verbose than have wildcards, even if it is burdensome.